### DISCUSSION TITLE
Plotting of coarse grained transition matrix

### DISCUSSION THREAD

**Discussion by WeilerP** [theislab/cellrank#719] [Ideas]
ATM, `GPCCA.plot_coarse_T` displays the coarse grained transition matrix for all macro-states. I am now wondering if it wouldn't be useful/better to only consider the terminal states (if they have been set)?

For example, I am running

```python
g = GPCCA(kernel)
g.compute_schur()
g.plot_spectrum(real_only=True)
g.compute_macrostates(n_states=5)
g.set_terminal_states_from_macrostates(["Terminal 1", "Terminal 2", "Terminal 3"])
g.compute_absorption_probabilities()
```

since more macro-states than actual terminal states have been identified. Following,

```python
g.plot_absorption_probabilities(same_plot=False, basis="umap", perc=[0, 99])
```

shows only the absorption probabilities towards the manually set terminal states. However,

```python
g.plot_coarse_T()
```

still plots the coarse grained transition probabilities for all macro-states found. If we subsetted to terminal states, won't all other macro-states have zero stationary distribution?

Wjat do you think @Marius1311, @michalk8?

**Comment by Marius1311** [comment]
Hi @WeilerP, this isn't going to work because the computation of the macrostates is linked to the computation of the macrostates, please see https://pubs.acs.org/doi/10.1021/acs.jctc.8b00079 and https://aip.scitation.org/doi/10.1063/1.5064530. 

It's a problem of re-distributing mass. Computing the coarse-grained transition matrix only works with well-defined macrostates, i.e. having a positive matrix with rows that sum to one. When you subset to some rows, how do you re-distribute mass away from the satats you removed towards the ones you left? This can have hughe effects on your results, I encourage you to test this on a simple example. @michalk8 have in the past discussed this in length and came to the conclusion that it makes no sense to subset macrostates and then compute a coarse-grained transition matrix among them, it just doesn't work.
